src/main/webapp/app/exercises/programming/manage/programming-exercise-detail.component.html

```diff
@@ -182,9 +182,8 @@ <h2><span jhiTranslate="artemisApp.programmingExercise.detail.title">Programming
                         <fa-icon [icon]="faExclamationTriangle" class="text-warning"></fa-icon>
                         <span jhiTranslate="artemisApp.programmingExercise.lockAllRepositories">Lock all repositories</span>
                     </button>
-
                     <button
-                        *ngIf="programmingExercise.isAtLeastEditor"
+                        *ngIf="programmingExercise.isAtLeastEditor && !programmingExercise.isAtLeastInstructor"
```

Comment:
Why should instructors not have this option?
Is this only one of the reset options in jhi-programming-exercise-reset? If so, you should handle it there and reduce the duplication and unify similar components and behavior there

Reply:
Hmm, so you mean that I should just show the `Reset Programming Exercise` dialog with only the `recreate build plans` option available? All other options are unavailable for the Editor. I thought it would be nicer to just have the recreate build plans button right there instead of hidden in a dialog under reset.

Reply:
I See advantages for both sides:
Easier code with less duplication and a more unified UI (e.g. an instructor is only an editor in another course) vs.
Simpler UI for editors.
Feel free to choose one of the options